### PR TITLE
feat: products listing page with filters

### PR DESCRIPTION
**Pull request by shraddha15-creator**
Added product listing page with filters
 
- Filter
- sort by
- - Low-to-high
- - High-to-low
- Price Range
- Filter by types
- filter by Ratings

PRODUCT-LISTING-WITH-FILTERS

### CODE REVIEW

File: src/pages/ProductListing/Products.js [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+				setProducts(reponse.data.products);
+			} catch (error) {
+				console.error("ERROR: While fetching Products", error);
+			}
+		})();
+	}, []);
+
+	const sortedProducts = sortFunction(products, sortBy);
+	const filterdProducts = filterFunction(sortedProducts, categories);
+	const categoriesTypeProducts = categoriesFunction(
```
Comment (Ash-D23) [inline src/pages/ProductListing/Products.js:33]: suggestion you can move all this sort,filter functions inside another function which will in end return a filtered products

**Review by Ash-D23 — APPROVED** [review]
Code is well structured, liked the way context and reducer are used
Suggestion: unless you have the use of filter context with any other component you should avoid wrapping it around App, you can wrap the filter context around product listing page

File: src/App.css [hunk 2]
```diff
@@ -1,4 +1,6 @@
 @import url("https://unfold-ui.netlify.app/css/components.css");
+@import url('https://fonts.googleapis.com/css2?family=Poppins:wght@100;200;300;400;500;600;700;800;900&display=swap');
+
```
Comment (devdeadviz) [inline src/App.css:3]: Add all your styles imports in `index.html` file.

**Review by devdeadviz — APPROVED** [review]
The code is clean and well structured. Good work!

**Comment by shraddha15-creator** [comment]
> Code is well structured, liked the way context and reducer are used Suggestion: unless you have the use of filter context with any other component you should avoid wrapping it around App, you can wrap the filter context around product listing page

sure will do that. Thank you so much for your suggestion.

File: src/App.css (hunk 2 above)
Comment (shraddha15-creator) [reply src/App.css:3]: > Add all your styles imports in `index.html` file.

Sure I will do that

**Comment by shraddha15-creator** [comment]
> The code is clean and well structured. Good work!

Thank you for review.

**Pull request merged**